core/src/main/scala/kafka/message/MessageSet.scala

```diff
@@ -82,6 +107,34 @@ abstract class MessageSet extends Iterable[MessageAndOffset] {
   def sizeInBytes: Int
 
   /**
+   * Convert this message set to use specified message format.
+   */
+  def toMessageFormat(toMagicValue: Byte): ByteBufferMessageSet = {
```

Comment:
It seems that we only need to convert message format on FileMessageSet. So, perhaps it's better to move this to FileMessageSet?